PacketDefinitions420/PacketHandlerManager.cs

```diff
@@ -258,17 +259,32 @@ public bool BroadcastPacket(byte[] data, Channel channelNo, PacketFlags flag = P
 
         public bool HandlePacket(Peer peer, byte[] data, Channel channelId)
         {
-            var header = new PacketHeader(data);
-            var convertor = GetConvertor(header.Cmd, channelId);
+            var reader = new BinaryReader(new MemoryStream(data));
+            RequestConvertor convertor;
 
-            switch (header.Cmd)
+            if (channelId == Channel.CHL_COMMUNICATION || channelId == Channel.CHL_LOADING_SCREEN)
             {
-                case GamePacketID.World_SendCamera_Server:
-                case GamePacketID.Waypoint_Acc:
-                case GamePacketID.OnReplication_Acc:
-                    break;
+                var loadScreenPacketId = (LoadScreenPacketID)reader.ReadByte();
+
+                convertor = GetConvertor(loadScreenPacketId);
+            }
+            else
+            {
+                var gamePacketId = (GamePacketID)reader.ReadByte();
+
+                convertor = GetConvertor(gamePacketId, channelId);
+
+                switch (gamePacketId)
+                {
+                    case GamePacketID.World_SendCamera_Server:
+                    case GamePacketID.Waypoint_Acc:
+                    case GamePacketID.OnReplication_Acc:
+                        break;
```

Comment:
Actually, I don't think this code is doing anything. Will open issue about that.